shim/integration/kvm_get_irqchip.cpp

```diff
@@ -0,0 +1,56 @@
+/// @copyright
+/// Copyright (C) 2020 Assured Information Security, Inc.
+///
+/// @copyright
+/// Permission is hereby granted, free of charge, to any person obtaining a copy
+/// of this software and associated documentation files (the "Software"), to deal
+/// in the Software without restriction, including without limitation the rights
+/// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+/// copies of the Software, and to permit persons to whom the Software is
+/// furnished to do so, subject to the following conditions:
+///
+/// @copyright
+/// The above copyright notice and this permission notice shall be included in
+/// all copies or substantial portions of the Software.
+///
+/// @copyright
+/// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+/// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+/// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+/// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+/// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+/// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+/// SOFTWARE.
+
+#include <integration_utils.hpp>
+#include <ioctl_t.hpp>
+#include <shim_platform_interface.hpp>
+
+#include <bsl/convert.hpp>
+#include <bsl/enable_color.hpp>
+#include <bsl/exit_code.hpp>
+#include <bsl/safe_integral.hpp>
+
+/// <!-- description -->
+///   @brief Provides the main entry point for this application.
+///
+/// <!-- inputs/outputs -->
+///   @return bsl::exit_success on success, bsl::exit_failure otherwise.
+///
+[[nodiscard]] auto
+main() noexcept -> bsl::exit_code
+{
+    bsl::enable_color();
+    integration::ioctl_t mut_system_ctl{shim::DEVICE_NAME};
+    auto const vmfd{mut_system_ctl.send(shim::KVM_CREATE_VM)};
+    integration::ioctl_t mut_vm{bsl::to_i32(vmfd)};
+    auto const vcpufd{mut_vm.send(shim::KVM_CREATE_VCPU)};
+    integration::ioctl_t mut_vcpu{bsl::to_i32(vcpufd)};
+    constexpr auto mut_ret{0_i64};
+    {
+        auto const irqchip{mut_vcpu.send(shim::KVM_GET_IRQCHIP)};
```

Comment:
CREATE_IRQCHIP will be implemented in another branch

Reply:
The problem is, this integration test is still wrong.

Reply:
Could you please provide more details "this integration test is still wrong"? is not sufficient. As I mentioned first we should complete the review of the create IRQCHIP and then we can update this get/set IRQCHIP accordingly.

Reply:
Please read my review about the wide contract [here](https://github.com/Bareflank/MicroV/pull/115/files#r806037865) as this will need it to.

Also, the `KVM_GET_IRQCHIP` ioctl here is made against a `vcpu` fd, but it should be made against a `vm` fd like you did in the other integration test for the `KVM_SET_IRQCHIP` ioctl. Thanks.

Reply:
yes, it is corrected now.